plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt

```diff
@@ -460,6 +460,65 @@ class LiquidDataflowTests {
       withoutPlugin = { compiles }
     )
   }
+
+  @Test
+  fun `subtyping, pre- and post-conditions are checked in overridden functions`() {
+    """
+      ${imports()}
+      open class A() {
+        open fun f(): Int = 2.post({ it > 0 }) { "greater than 0" }
+      }
+      
+      class B(): A() {
+        override fun f(): Int = 0
+      }
+      """(
+      withPlugin = { failsWith { it.contains("`f` fails to satisfy the post-condition") } },
+      withoutPlugin = { compiles }
+    )
+  }
+
+  @Test
+  fun `subtyping, Liskov Substitution Principle, 1`() {
```

Comment:
Since this is related to subtyping, we should perhaps include tests for `object`, `interface`, `fun interface`, and anonymous inline implementations. Another question is generic and type-bound restrictions in the function declarations. For example, `fun <A: Comparable> A....` the Liskov substitution principle should also apply in these cases since the target functions can be refined by predicates carried in the type bounds.

Reply:
One question there is: how should be attach laws to things without a body, such as members in an `interface`?

Reply:
For cases where they actually have a body, it would be the same, interfaces in Kotlin allow implemented functions.
For cases like interfaces or abstract classes where there is nobody all I can think at the moment is using the annotations directly or allowing a special kind of synthetic body that is just used for constraint declaration and that the compiler plugin removes to ensure the member remains abstract.